parquet/api.py

```diff
@@ -0,0 +1,280 @@
+        """
+        out = {}
+        fname = self.fname
+        helper = schema.SchemaHelper(self.schema)
+        if filters and filter_out_stats(rg, filters, helper):
+            return pd.DataFrame()
+        if filters and self.cats and filter_out_cats(rg, filters):
+            return pd.DataFrame()
+
+        for col in rg.columns:
+            name = ".".join(col.meta_data.path_in_schema)
+            se = helper.schema_element(name)
+            if name not in cols:
+                continue
+
+
+            if col.file_path is None:
+                # continue reading from the same base file
+                infile = self.f
+            else:
+                # relative file
+                ofname = os.path.join(os.path.dirname(self.fname),
+                                      col.file_path)
+                if ofname != fname:
+                    # open relative file, if not the current one
+                    infile = self.open(ofname)
+                    fname = ofname
+
+            use = name in usecats if usecats is not None else False
+            s = core.read_col(col, helper, infile, use_cat=use)
+            out[name] = s
+        out = pd.DataFrame(out)
+
+        # apply categories
+        for cat in self.cats:
+            # *Hard assumption*: all chunks in a row group have the
+            # same partition (correct for spark/hive)
+            partitions = re.findall("([a-zA-Z_]+)=([^/]+)/",
+                                    rg.columns[0].file_path)
+            val = [p[1] for p in partitions if p[0] == cat][0]
+            codes = np.empty(rg.num_rows, dtype=np.int16)
+            codes[:] = self.cats[cat].index(val)
+            out[cat] = pd.Categorical.from_codes(
+                    codes, [val_to_num(c) for c in self.cats[cat]])
+        return out
+
+    def to_pandas(self, columns=None, usecats=None, filters={}):
+        cols = columns or self.columns
+        tot = [self.read_row_group(rg, cols, usecats, filters=filters) for rg in
+               self.row_groups]
+        tot = [t for t in tot if len(tot)]
+
+        # TODO: if categories vary from one rg to next, need
+        # pandas.types.concat.union_categoricals
+        return pd.concat(tot, ignore_index=True)
+
+    @property
+    def count(self):
+        return sum(rg.num_rows for rg in self.row_groups)
+
+    @property
+    def info(self):
+        return {'name': self.fname, 'columns': self.columns,
+                'categories': list(self.cats), 'rows': self.count}
+
+    @property
+    def dtypes(self):
+        dtype = {f.name: converted_types.typemap(f)
+                 for f in self.schema if f.num_children is None}
+        for cat in self.cats:
+            dtype[cat] = "category"
+            # pd.Series(self.cats[cat]).map(val_to_num).dtype
+        return dtype
+
+    @property
+    def divisions(self):
+        return np.cumsum([0] + [rg.num_rows for rg in self.row_groups])
```

Comment:
I don't think we should publish this property.  

Eventually I think we'll have to look through metadata, find columns that are sorted, and offer them up as the index with divisions.  Providing a numerical row-index divisions is rarely useful.